runtime/metricsview/executor_rewrite_druid_groups.go

```diff
@@ -22,6 +22,20 @@ func (e *Executor) rewriteDruidGroupsWalk(n *SelectNode) error {
 		return nil
 	}
 
+	// Recurse over FromSelect as it may contain sub queries with joins
+	if n.FromSelect != nil {
+		err := e.rewriteDruidGroupsWalk(n.FromSelect)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Now there are two "// Recurse ..." sections in this function – what about just moving the `// Recurse` section above the `// Rewrite` section, and having a `var hasJoins bool` that gets set in the join recurse cases?